crates/stages/stages/src/stages/sender_recovery.rs

```diff
@@ -153,23 +153,29 @@ where
         .unzip();
 
     let static_file_provider = provider.static_file_provider().clone();
-    tokio::task::spawn_blocking(move || {
+    std::thread::spawn(move || {
```

Comment:
this should be fine and overhead is negligible since our batches are very large, but could be optimized away by spawning one thread for this for all batches that receives the chunks from every batch